### PR TITLE
update to 1.20.3

### PR DESCRIPTION
**Pull request by arnokeesman**
not tested beyond joining and going afk

gonna need to create a 1.20.2 branch as well when we get to merging

### CODE REVIEW

File: ec-core/src/main/java/dev/jpcode/eccore/config/ConfigUtil.java [hunk 1]
```diff
@@ -57,16 +72,15 @@ public static Style parseStyle(String styleStr) {
...
         if (outStyle == null) {
             try {
-                outStyle = STYLE_JSON_DESERIALIZER.deserialize(
-                    JsonParser.parseString(styleStr),
-                    null, null
-                );
+                outStyle = Style.Codecs.CODEC.parse(
+                    JsonOps.INSTANCE,
+                    JsonHelper.deserialize(styleStr)
+                ).result().orElse(Style.EMPTY);
```
Comment (arnokeesman) [inline ec-core/src/main/java/dev/jpcode/eccore/config/ConfigUtil.java:78]: should the orElse be `null`?

File: src/main/java/com/fibermc/essentialcommands/playerdata/PlayerData.java
Comment (arnokeesman) [inline src/main/java/com/fibermc/essentialcommands/playerdata/PlayerData.java:1]: parsing "null" in text serialization throws a jsonformat exception
is this an okay way to handle it?

**Review by arnokeesman — COMMENTED** [review]
here what I consider needs some attention before it's ready to merge

**Comment by John-Paul-R** [comment]
Will address remaining things post-merge

**Pull request merged**